.Net: Add AssemblyAI file service Remake from #5964

RogerBarreto opened the pull request:
### Motivation and Context

<!-- Thank you for your contribution to the semantic-kernel repo!
Please help reviewers and future users, providing the following information:
  1. Why is this change required?
  2. What problem does it solve?
  3. What scenario does it contribute to?
  4. If it fixes an open issue, please link to the issue here.
-->

### Description

<!-- Describe your changes, the overall approach, the underlying design.
     These notes will help understanding how your code works. Thanks! -->

### Contribution Checklist

<!-- Before submitting this PR, please make sure: -->

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

RogerBarreto commented on dotnet/src/Connectors/Connectors.AssemblyAI/Files/AssemblyAIFileService.cs:
```diff
@@ -0,0 +1,55 @@
+    /// <param name="stream">The file stream</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>The file metadata.</returns>
+    public async Task<AudioContent> UploadAsync(Stream stream, CancellationToken cancellationToken = default)
```
```suggestion
    public async Task<AudioContent> UploadAsync(Stream stream, CancellationToken cancellationToken = default)
```
@Swimburger Consider changing the signature of this service to use `IAsyncEnumerable<byte[]>` as the input.

This will avoid `IDisposable` contents to be sent to the service and being managed by both caller / internals.

Swimburger replied:
I don't have permission to update your PR. 

Either way, I only added this remark to the `UploadAsync` method:
```csharp
    /// <remarks>
    /// Caller responsible for disposal of the stream.
    /// </remarks>
```

Since the point of this service is to upload files, the expected type used should be `Stream`.
When users pass a stream to a method, it is expected by precedence that the `Stream` should be disposed by the caller, unless there's an additional parameter that lets you control this behavior.

I considered using `BinaryContent` but the implementation of retrieving the content from this object can potentially duplicate the memory.

RogerBarreto commented:
@Swimburger, added minor changes to overall code.

- Stopped using `JsonDocument` in favor of `JsonElement` for performance
- Changed structure from `Files` to `Services`, all public services should be in the root namespace.
- Changes `Client` to be `Core` in line with our other connectors.
- Update Namespaces on UnitTests.
- Sealing specialized class.

Swimburger commented:
Those changes look great. I didn't know about the perf difference, thanks for sharing and updating!
